### PR TITLE
HAAR-878 Add ability to get assignable roles for a user

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/ExternalUserRolesController.kt [hunk 1]
```diff
@@ -130,6 +128,56 @@ class ExternalUserRolesController(
     externalUserRolesService.removeRoleByUserId(userId, role)
     log.info("Remove role succeeded for userId {} and role code {}", userId, role)
   }
+
+  @PreAuthorize("hasAnyRole('ROLE_MAINTAIN_OAUTH_USERS', 'ROLE_AUTH_GROUP_MANAGER')")
+  @GetMapping("{userId}/assignable-roles")
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/ExternalUserRolesController.kt:133]: I've added on these role s- but not sure if they're needed as we don't have any on external users.  Can remove at a later date if decide not

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/ExternalUserRolesController.kt [hunk 2]
```diff
@@ -130,6 +128,56 @@ class ExternalUserRolesController(
...
+    description = "Roles for a specific user. Requires role ROLE_MAINTAIN_ACCESS_ROLES_ADMIN or ROLE_MAINTAIN_ACCESS_ROLES",
+    security = [SecurityRequirement(name = "MAINTAIN_ACCESS_ROLES"), SecurityRequirement(name = "MAINTAIN_ACCESS_ROLES_ADMIN")],
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "User role list"
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Incorrect request to get caseloads for a user",
```
Comment (AlTurner-MOJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/ExternalUserRolesController.kt:146]: Description refers to caseloads?

**Review by AlTurner-MOJ — APPROVED** [review]
lgtm

**Review by AlTurner-MOJ — APPROVED** [review]
lgtm

**Pull request merged**